packages/form-footer/src/FormFooter.tsx

```diff
@@ -20,97 +22,92 @@ import PrimaryButton, { PrimaryButtonProps } from './PrimaryButton';
-              darkMode={darkMode}
-              className={bannerStyle}
-              variant="danger"
-            >
-              {errorMessage}
-            </Banner>
-          )}
-          {showCancelButton && (
+    <LeafyGreenProvider darkMode={darkMode}>
+      <footer
+        data-testid="lg-form_footer-footer"
+        className={cx(footerBaseStyle, footerThemeStyle[theme], className)}
+        {...rest}
+      >
+        <div className={cx(contentStyle, contentClassName)}>
+          {showBackButton && (
             <Button
               variant="default"
-              onClick={_cancelButtonProps.onClick}
-              className={buttonStyle}
-              leftGlyph={_cancelButtonProps.leftGlyph}
-              darkMode={darkMode}
-              data-testid="lg-form_footer-cancel_button"
+              leftGlyph={
+                <ArrowLeftIcon data-testid="lg-form_footer-back_button-icon" />
+              }
+              data-testid="lg-form_footer-back_button"
+              {...backButtonProps}
             >
-              {_cancelButtonProps.children || 'Cancel'}
+              {backButtonProps?.children || 'Back'}
             </Button>
           )}
-          {isComponentType(primaryButton as React.ReactElement, 'Button') ? (
-            React.cloneElement(primaryButton as React.ReactElement, {
-              darkMode: darkMode,
-              ['data-testid']: 'lg-form_footer-primary_button',
-            })
-          ) : (
-            <PrimaryButton
-              darkMode={darkMode}
-              data-testid="lg-form_footer-primary_button"
-              {...(primaryButton as PrimaryButtonProps)}
-            />
-          )}
+          <div className={flexEndContent}>
+            {errorMessage && (
+              <Banner className={bannerStyle} variant="danger">
+                {errorMessage}
+              </Banner>
+            )}
+            {showCancelButton && (
+              <Button
+                data-testid="lg-form_footer-cancel_button"
+                {...cancelButtonProps}
+                variant="default"
+              >
+                {cancelButtonProps?.children || 'Cancel'}
+              </Button>
+            )}
+            {showDeprecatedPrimaryButton ? (
+              isComponentType(primaryButton as React.ReactElement, 'Button') ? (
+                React.cloneElement(primaryButton as React.ReactElement, {
+                  darkMode: darkMode,
+                  ['data-testid']: 'lg-form_footer-primary_button',
+                })
+              ) : (
+                <PrimaryButton
+                  darkMode={darkMode}
+                  data-testid="lg-form_footer-primary_button"
+                  {...(primaryButton as PrimaryButtonProps)}
+                />
+              )
+            ) : (
+              <Button
+                variant="primary"
+                data-testid="lg-form_footer-primary_button"
+                {...primaryButtonProps}
+              />
```

Comment:
wonder if this can also use the existing `PrimaryButton` or at least apply the same approach to defining the variant prop. It seems more explicit that `variant` is overwritten whereas this current approach makes it seem like it's hardcoded to `"primary"`

Reply:
I like this approach because it avoids destructuring and extra lines of code. I don't think less explicit is a bad thing since it's still going to be overwritten.